src/Parsers/ASTAlterNamedCollectionQuery.cpp

```diff
@@ -12,41 +11,39 @@ ASTPtr ASTAlterNamedCollectionQuery::clone() const
     return std::make_shared<ASTAlterNamedCollectionQuery>(*this);
 }
 
-void ASTAlterNamedCollectionQuery::formatImpl(const IAST::FormatSettings & settings, IAST::FormatState &, IAST::FormatStateStacked) const
+void ASTAlterNamedCollectionQuery::formatImpl(FormattingBuffer out) const
 {
-    settings.ostr << (settings.hilite ? hilite_keyword : "") << "Alter NAMED COLLECTION ";
-    settings.ostr << (settings.hilite ? hilite_identifier : "") << backQuoteIfNeed(collection_name) << (settings.hilite ? hilite_none : "");
-    formatOnCluster(settings);
+    out.writeKeyword("Alter NAMED COLLECTION ");
+    out.writeProbablyBackQuotedIdentifier(collection_name);
```

Comment:
This is quite a long function name. Can it be shorter? And also "probably" doesn't sound very good (like "probably" or "probably not"?) "If needed" looks better

Reply:
The name is due to the function name which was used in place of all the usages of my newly introduced `writeProbablyBackQuotedIdentifier`.

https://github.com/murfel/ClickHouse/blob/69957e214762b884d759dc0169b87745367d1227/src/Parsers/IAST.cpp#L409

For consistency, I won't be fixing this.

But I was also confused by the numerous `write*` functions, and `probably` didn't help :(

Reply:
ok